### DISCUSSION TITLE
[General] How to spend from L2 account in a force included transaction

### DISCUSSION THREAD

**Discussion by NIC619** [zkSync-Community-Hub/zksync-developers#523] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I'm studying the force inclusion mechanism on zkSync. I ran into a problem where I want to force include a transaction that spends from my L2 account, e.g., transfer my L2 ETH to another account, but L1's `MailBox.requestL2Transaction` function requires that I pay up the `l2Value` immediately instead of spending from my L2 account.
https://github.com/matter-labs/era-contracts/blob/4aa7006153ad571643342dff22c16eaf4a70fdc1/l1-contracts/contracts/zksync/facets/Mailbox.sol#L251

How am I supposed to spend from my L2 account if I'm being censored by the sequencer?

**Comment by bxpana** [comment]
Appreciate the post. Just to make sure we're on the same page, where are you reading about the forced inclusion on zkSync? Want to make sure I'm not already sending you resources you've already checked out.

**Reply by NIC619** [reply]
I began with [l2beat](https://l2beat.com/scaling/projects/zksync-era?ref=bankless.ghost.io#force-transactions) and then it's reference link to [L1/L2 interop doc](https://docs.zksync.io/build/developer-reference/l1-l2-interop.html) and [L1->L2 tx tutorial](https://docs.zksync.io/build/tutorials/how-to/send-transaction-l1-l2.html). Next I go into [Mailbox contract](https://github.com/matter-labs/era-contracts/blob/4aa7006153ad571643342dff22c16eaf4a70fdc1/l1-contracts/contracts/zksync/facets/Mailbox.sol#L192) and [Executor contract](https://github.com/matter-labs/era-contracts/blob/4aa7006153ad571643342dff22c16eaf4a70fdc1/l1-contracts/contracts/zksync/facets/Executor.sol#L193-L196).

In the end I [tested](https://etherscan.io/tx/0xe3d78d10e0f351638ad5ae65cd07ddd489d5e7790b351ba365a2d0fe67429c20) the [force inclusion](https://era.zksync.network/tx/0x4e0dcfef1cdfd6aabbb52baad93c7b4881e0e0b91391ff86df3501e346c8ec2d) myself but found that I couldn't spend from my L2 account.

**Reply by bxpana** [reply]
Got it. Am I understanding correctly that you are trying to force a transaction on L2 from L1 that is not a withdrawal?

**Reply by NIC619** [reply]
From the perspective of a user that's trying to use the force inclusion mechanism, it probably does not matter much whether it's a withdrawal or not. If a user is being censored and wants out, then he will force include a withdrawal transaction for sure but it could also be that the sequencer is going offline and the user still need to transact.

I intend to try both: (1) force include a transaction that is not a withdrawal and (2) force include a transaction that is a withdrawal.

**Reply by NIC619** [reply]
Hi @bxpana , is there any update about how I can spend from L2 account using force inclusion? For example, in Optimism's force inclusion user specifies L2 ETH value `_value` to spend from the L2 account while it also provides the `msg.value` option that allows user to deposit from L1:
https://github.com/ethereum-optimism/optimism/blob/a3cc8f275d1d4a20bcdddfe7b421b3b34bbf0785/packages/contracts-bedrock/src/L1/OptimismPortal.sol#L409

**Reply by bxpana** [reply, marked as answer]
Seems this is a known issue and something that we are working on.
